ultraviolet/shared/src/main/scala/ultraviolet/macros/CreateShaderAST.scala

```diff
@@ -1236,6 +1236,56 @@ class CreateShaderAST[Q <: Quotes](using val qq: Q) extends ShaderMacroUtils:
           case _ =>
             throw ShaderError.Unsupported("Shaders do not support infix operator: " + op)
 
+      case Apply(
+            Apply(Ident(op), List(Apply(_, List(Typed(Repeated(List(Literal(StringConstant(value))), _), _))))),
+            _
+          ) =>
+        op match
+          case "hex" =>
+            import ultraviolet.syntax.interpolators.hex.*
+            value.toVec3.map(v => ShaderAST.DataTypes.vec3(v.x, v.y, v.z)).getOrElse {
+              throw ShaderError.Unsupported(
+                "Shaders do not support hex colours that are not 6 characters long (expected format '#FF00FF')."
```

Comment:
I think the error could be slightly more helpful, so that they don't get this message and assume 8 char hex codes aren't an option.

Maybe something like:
"Hex values must be 6 characters or 8 characters long using the `hex` or `hexa` interpolators, respectively."

You could use the same error on `hexa`, and something similar for the rgb/rgba ones. What do you think?